### PR TITLE
[CSL-1976] Add tracking to components

### PR DESCRIPTION
**Pull request by esezen**
- Quiz Results Load
- Quiz Result Click
- Quiz Conversion

### CODE REVIEW

File: src/components/ResultCard/ResultCard.tsx [hunk 1]
```diff
@@ -1,31 +1,71 @@
...
+  const { cioClient } = useContext(QuizContext);
   const salePrice = salePriceKey && result?.data?.[salePriceKey];
   const regularPrice = regularPriceKey && result?.data?.[regularPriceKey];
 
-  const clickHandler = () => {
-    if (clickItemCallback && typeof clickItemCallback === 'function') {
-      clickItemCallback(result);
+  const clickItemCallback = () => {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    if (resultsResponse && resultsResponse.request && resultsResponse.response) {
```
Comment (esezen) [inline src/components/ResultCard/ResultCard.tsx:31]: This code partially repeats. I thought about moving the shared parts to a separate function but I think there is enough difference that makes it enough to keep this as is. Let me know what you guys think

File: src/components/ResultCard/ResultCard.tsx [hunk 2]
```diff
@@ -1,31 +1,71 @@
...
+  const {
+    result,
+    addToCartCallback,
+    clickItemCallback: customClickItemCallback,
+    salePriceKey,
+    regularPriceKey,
+    resultPosition,
+  } = props;
+  const { resultsResponse } = useContext(QuizContext);
+  const { cioClient } = useContext(QuizContext);
```
Comment (stanlp1) [inline src/components/ResultCard/ResultCard.tsx:25]: 🤔 Is there any reason why we have these on separate lines?

File: src/components/ResultContainer/ResultContainer.tsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ export default function ResultContainer(props: IResultContainerProps) {
     resultCardRegularPriceKey,
   } = options;
   const { resultsResponse } = useContext(QuizContext);
+  const { cioClient } = useContext(QuizContext);
```
Comment (stanlp1) [inline src/components/ResultContainer/ResultContainer.tsx:23]: Same as here, why do we have these on separate lines? Am I missing something 🤔

File: src/components/ResultCard/ResultCard.tsx [hunk 4]
```diff
@@ -49,24 +89,20 @@ export default function ResultCard(props: ResultCardProps) {
     </>
```
Comment (stanlp1) [inline src/components/ResultCard/ResultCard.tsx:89]: Should we pass price to the ResultCtaButton?

**Review by stanlp1 — CHANGES_REQUESTED** [review]
Minor comments, implementation looks good to me.

File: src/components/ResultCard/ResultCard.tsx (hunk 2 above)
Comment (esezen) [reply src/components/ResultCard/ResultCard.tsx:25]: Not really 🤔 Similar code was on separate lines on existing code so I didn't really think why and followed that. I think they can be on the same line. Will update

File: src/components/ResultCard/ResultCard.tsx (hunk 4 above)
Comment (esezen) [reply src/components/ResultCard/ResultCard.tsx:89]: Good catch. I had it then I accidentally removed it when I was reorganizing code here [6e20c11](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/56/commits/6e20c117e4144c3d33185457edb5ff5b4d5015c8)

**Review by stanlp1 — APPROVED** [review]
Nice lgtm.

**Pull request merged**